ocaml/lambda/matching.ml

```diff
@@ -100,25 +99,25 @@ open Printpat
 module Scoped_location = Debuginfo.Scoped_location
 
 type error =
-    Non_value_layout of Layout.Violation.t
-  | Illegal_record_field of Layout.const
+    Non_value_jkind of Jkind.Violation.t
+  | Illegal_record_field of Jkind.const
 
 exception Error of Location.t * error
 
 let dbg = false
 
 (* CR layouts v5: When we're ready to allow non-values, these can be deleted or
    changed to check for void. *)
-let layout_must_be_value loc layout =
-  match Layout.(sub layout (value ~why:V1_safety_check)) with
+let jkind_must_be_value loc jkind =
```

Comment:
Similarly I think this check is really about layouts.

Reply:
This is debatable. The thing we care about is whether the layout is value, as you say. But the argument to this function must remain a jkind, because that's what we have to pass it. That all might change, but building up that distinction is going to happen in future PRs; I do not want to implement that in this one. I'll see if I can find a better name.

Reply:
(I'm convinced, no need to find a better name.)